src/layout/rect.rs

```diff
@@ -86,6 +116,75 @@ impl Rect {
         }
     }
 
+    /// Adjusts the size of the `Rect`.
+    ///
+    /// Adjust the size of the `Rect` by moving either the top or left. This effectively changes
+    /// [`Rect::x`] and [`Rect::y`] and recalculate [`Rect::width`] and [`Rect::height`] so that
+    /// the right and bottom "borders" do not move.
+    ///
+    /// - Positive `x` moves the left "border" to the right, negative to the left.
+    /// - Positive `y` moves the top "border" to the bottom, negative to the top.
+    ///
+    /// # Caveats
+    ///
+    /// If the given offset overflows the `Rect`'s fields, resulting values will wrapped around.
+    ///
+    /// ```
+    /// # use ratatui::layout::*;
+    /// let rect = Rect::new(0, 0, 10, 10).resize(Offset {top: 0, right: 0, bottom: 0, left: -1});
+    /// assert_eq!(rect, Rect::new(u16::MAX, 0, 11, 10));
+    /// ```
+    ///
+    /// # See also
+    ///
+    /// See also [`Rect::offset`] to only move the `Rect`.
+    pub fn resize<I>(self, offset: I) -> Rect
+    where
+        I: Into<Offset>,
+    {
+        let offset: Offset = offset.into();
+        Rect {
+            x: (self.x as i32 + offset.left) as u16,
+            y: (self.y as i32 + offset.top) as u16,
+            width: (self.width as i32 + offset.right - offset.left) as u16,
+            height: (self.height as i32 + offset.bottom - offset.top) as u16,
+        }
+    }
+
+    /// Moves the `Rect` without modifying its size.
```

Comment:
It would, but unfortunately it is a reserved keyword. To use it we would have to write `r#move` which is a bit cumbersome imo.

What do you think? What about `relocate`?

Reply:
Ahh, that's annoying. I think perhaps then we come full circle to `offset()` being the right name for this. Relocate seems like a bit of a clunky name for moving something, while `offset` seems to be found throughout ratatui already and means mostly the same as what we'd want this to mean.